staging/cse/windows/azurecnifunc.ps1

```diff
@@ -66,19 +66,23 @@ function Set-AzureCNIConfig
         $configJson.plugins[0].AdditionalArgs[0] = $jsonContent
 
         # TODO: Remove it after Windows OS fixes the issue.
-        Write-Log "Update RegKey to disable the incompatible HNSControlFlag (0x10) for feature DisableWindowsOutboundNat"
-        $hnsControlFlag=0x10
-        $currentValue=(Get-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Services\hns\State" -Name HNSControlFlag -ErrorAction Ignore)
-        if (![string]::IsNullOrEmpty($currentValue)) {
-            Write-Log "The current value of HNSControlFlag is $currentValue"
-            # Set the bit to 0 if the bit is 1
-            if ([int]$currentValue.HNSControlFlag -band $hnsControlFlag) {
-                $hnsControlFlag=([int]$currentValue.HNSControlFlag -bxor $hnsControlFlag)
-                Set-ItemProperty -Path "HKLM:\SYSTEM\CurrentControlSet\Services\hns\State" -Name HNSControlFlag -Type DWORD -Value $hnsControlFlag
+        # WS2022 has fix, so only need to update registry for WS2019.
+        $osBuildNumber = (get-wmiobject win32_operatingsystem).BuildNumber
+        if ($osBuildNumber -le 17763){

```

Comment:
Use Get-WindowsVersion?

Reply:
It seems like that there is another place with same code in Set-AzureCNIConfig
Two options:
1. Merget this PR and update them to use Get-WindowsVersion
2. Update this in this PR and update the code in Set-AzureCNIConfig in another PR